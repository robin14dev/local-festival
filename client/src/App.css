@import url(./styles/reset.css);
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

:root {
  --primaryBlue: rgb(64 82 255);
  --primaryGreen: #60ff53;
  --primaryOrange: #ff9a62;
  --primaryPurple: #6268ff;
  --mainColor: var(--primaryPurple);
  --primaryPink: #fb1039;
  --lightBlack: #000000b3;
  --rem: 16;
}

button {
  background: inherit;
  border: none;
  box-shadow: none;
  border-radius: 0;
  padding: 0;
  overflow: visible;
  cursor: pointer;
}

/* input 기본 스타일 초기화 */
input {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  outline: none;
  border: none;
}

textarea {
  border: none;
  resize: none;
}

textarea:focus,
input:focus {
  outline: none;
}

.container {
  opacity: 1;
  background-color: yellow;
  transition: all 0.3s ease-out;
}

.container.unmount {
  opacity: 0;
}

@font-face {
  font-family: 'EarlyFontDiary';
  /* src: url('https://cdn.jsdelivr.net/gh/projectnoonnu/noonfonts_220508@1.0/EarlyFontDiary.woff2') format('woff2'); */
  src: url('./fonts/EF_Diary.ttf');
  font-weight: normal;
  font-style: normal;
}

@font-face {
  font-family: 'ulsanjunggu';
  /* src: url('https://cdn.jsdelivr.net/gh/projectnoonnu/noonfonts_2206-02@1.0/ulsanjunggu.woff2') format('woff2'); */
  src: url('./fonts/ulsanjunggu.ttf');
  font-weight: normal;
  font-style: normal;
}

@font-face {
  font-family: 'HS-Regular';
  src: url('https://cdn.jsdelivr.net/gh/projectnoonnu/noonfonts_2201-2@1.0/HS-Regular.woff')
    format('woff');
  font-weight: normal;
  font-style: normal;
}

@font-face {
  font-family: 'GangwonEdu_OTFBoldA';
  src: url('https://cdn.jsdelivr.net/gh/projectnoonnu/noonfonts_2201-2@1.0/GangwonEdu_OTFBoldA.woff')
    format('woff');
  font-weight: normal;
  font-style: normal;
}

@font-face {
  font-family: 'SuseongDotum';
  src: url('https://cdn.jsdelivr.net/gh/projectnoonnu/noonfonts_2205@1.0/SuseongDotum.woff2')
    format('woff2');
  font-weight: normal;
  font-style: normal;
}
@font-face {
  font-family: 'NanumSquareRound';
  src: url('https://cdn.jsdelivr.net/gh/projectnoonnu/noonfonts_two@1.0/NanumSquareRound.woff')
    format('woff');
  font-weight: normal;
  font-style: normal;
}

@keyframes toast-in-right {
  from {
    transform: translateX(100%);
  }
  to {
    transform: translateX(0);
  }
}

@keyframes button-loading-spinner {
  from {
    transform: rotate(0turn);
  }

  to {
    transform: rotate(1turn);
  }
}

@keyframes slide-down {
  from {
    transform: translateY(-1rem);
    opacity: 0;
  }

  to {
    opacity: 1;
    transform: translateY(0rem);
  }
}

@keyframes slide-up {
  from {
    transform: translateY(0rem);
    opacity: 1;
  }

  to {
    opacity: 0;
    transform: translateY(-1rem);
  }
}

@keyframes dark-soft {
  from {
    background: none;
  }

  to {
    background: rgba(0, 0, 0, 0.4);
  }
}
@keyframes bright-soft {
  from {
    background: rgba(0, 0, 0, 0.4);
  }

  to {
    background: none;
  }
}
