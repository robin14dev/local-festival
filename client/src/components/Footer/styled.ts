import styled from "styled-components";

export const Container = styled.footer`
  z-index: 10;
  width: 100%;
  height: 3rem;
  position: fixed;
  bottom: 0;
  left: 0;
  display: flex;
  border-top: 1px solid lightgray;
  justify-content: flex-start;
  background-color: #f1f3f5;
  color: black;
  padding: 0.5rem;
  img {
    width: 2rem;
    height: 2rem;
    margin-left: 1rem;
  }
  div {
    padding-left: 1rem;
    font-size: 0.7rem;
  }

  @media (max-width: 476px) {
    display: none;
  }
`;

export const MobileContainer = styled.footer`
  background-color: white;
  position: fixed;
  bottom: 0;
  z-index: 10;
  display: flex;
  justify-content: space-between;
  padding: 0 3rem;
  height: 4rem;
  width: 100%;
  @media (min-width: 476px) {
    display: none;
  }
`;

export const Item = styled.div`
  cursor: pointer;
  width: 4rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  img {
    height: 1.5rem;
    width: 1.5rem;
  }
  div {
    font-weight: 400;
    font-size: 12px;
  }
`;
