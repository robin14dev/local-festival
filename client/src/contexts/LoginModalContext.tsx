import LoginModal from "components/account/LoginModal";
import React, { useState, createContext } from "react";

type State = boolean;
type Dispatch = React.Dispatch<React.SetStateAction<State>>;

export const LoginModalStateContext = createContext<State>(false);

export const LoginModalDispatchContext = createContext<Dispatch>(() => {});

type Props = {
  children: JSX.Element | JSX.Element[];
};
export const LoginModalContextProvider = ({ children }: Props) => {
  const [isLoginModal, setIsLoginModal] = useState(false);
  return (
    <LoginModalStateContext.Provider value={isLoginModal}>
      <LoginModalDispatchContext.Provider value={setIsLoginModal}>
        {isLoginModal && <LoginModal/>}
        {children}
      </LoginModalDispatchContext.Provider>
    </LoginModalStateContext.Provider>
  );
};

/**
 * ModalContext로 해서 여러개 나올 수 있게 하는 방법 
 * 
 * showModal setShowModal 해서
 * 
 * 모달 나올때 키랑 컴포넌트 맵핑해서 보여주기 
 * 
 * 배열로 넣어서 렌더링
 * 
 * 
 * 
 */
