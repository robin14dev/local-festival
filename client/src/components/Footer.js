import React from 'react';
import peachmong from '../assets/peachmong.png';
import styled, { css } from 'styled-components';
import searchImage from '../assets/search-mobile.png';
import pickImage from '../assets/pick-mobile.png';
import profileImage from '../assets/profile-mobile.png';
import { useNavigate } from 'react-router-dom';
import { useCallback } from 'react';

const Wrapper = styled.div`
  width: 100%;
  height: 3rem;
  position: fixed;
  bottom: 0;
  display: flex;
  border-top: 1px solid lightgray;
  justify-content: flex-start;
  background-color: #f1f3f5;
  color: black;
  /* margin-top: 2rem; */
  padding: 0.5rem;
  /* box-shadow: 0 -2px 0 1px #f0f1f4; */
  img {
    width: 2rem;
    height: 2rem;
    margin-left: 1rem;
  }
  div {
    padding-left: 1rem;
    font-size: 0.7rem;
  }

  @media (max-width: 485px) {
    display: none;
  }
`;

const WrapperMobile = styled.div`
  background-color: white;
  position: fixed;
  bottom: 0;
  z-index: 10;
  display: flex;
  justify-content: space-between;
  padding: 0 3rem;
  height: 3.438rem;
  width: 100%;
  @media (min-width: 486px) {
    display: none;
  }
`;

const Item = styled.div`
  width: 4rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  img {
    height: 1.5rem;
    width: 1.5rem;
  }
  div {
    font-weight: 400;
    font-size: 12px;
  }
`;

const Footer = ({ authState, setLoginModal }) => {
  let navigate = useNavigate();
  const goPage = useCallback(
    (path) => {
      if (authState.loginStatus) {
        navigate(`/${path}`);
      } else {
        setLoginModal(true);
      }
    },
    [authState]
  );

  return (
    <>
      <Wrapper>
        <img src={peachmong} alt="peachmong" />
        <div>Copyright © 2022 Peachmong All rights reserved.</div>
      </Wrapper>
      <WrapperMobile>
        <Item onClick={() => navigate('/')}>
          <img src={searchImage} alt="home"></img>
          <div>둘러보기</div>
        </Item>
        <Item path="MyPick" onClick={() => goPage('MyPick')}>
          <img src={pickImage} alt="mypick"></img>
          <div>위시리스트</div>
        </Item>
        <Item onClick={() => goPage('AccountSetting')}>
          <img src={profileImage} alt="profile"></img>
          <div>프로필</div>
        </Item>
      </WrapperMobile>
    </>
  );
};

export default Footer;
