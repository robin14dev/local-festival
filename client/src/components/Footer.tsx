import React from 'react';
import peachmong from '../assets/peachmong.png';
import styled from 'styled-components';

import { useNavigate } from 'react-router-dom';
import { useCallback } from 'react';
import { ReactComponent as Account } from '../assets/profile.svg';
import { ReactComponent as Wishlist } from '../assets/heart-empty.svg';
import { ReactComponent as Main } from '../assets/search.svg';
const Wrapper = styled.footer`
  z-index: 10;
  width: 100%;
  height: 3rem;
  position: fixed;
  bottom: 0;
  left: 0;
  display: flex;
  border-top: 1px solid lightgray;
  justify-content: flex-start;
  background-color: #f1f3f5;
  color: black;
  padding: 0.5rem;
  img {
    width: 2rem;
    height: 2rem;
    margin-left: 1rem;
  }
  div {
    padding-left: 1rem;
    font-size: 0.7rem;
  }

  @media (max-width: 476px) {
    display: none;
  }
`;

const WrapperMobile = styled.footer`
  background-color: white;
  position: fixed;
  bottom: 0;
  z-index: 10;
  display: flex;
  justify-content: space-between;
  padding: 0 3rem;
  height: 4rem;
  width: 100%;
  @media (min-width: 476px) {
    display: none;
  }
`;

const Item = styled.div`
  cursor: pointer;
  width: 4rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  img {
    height: 1.5rem;
    width: 1.5rem;
  }
  div {
    font-weight: 400;
    font-size: 12px;
  }
`;

type FooterProps = {
  authState: AuthState;
  setIsLoginModal: React.Dispatch<React.SetStateAction<boolean>>;
};

const Footer = ({ authState, setIsLoginModal }: FooterProps) => {
  let navigate = useNavigate();
  const goPage = useCallback(
    (path: string) => {
      if (authState.loginStatus) {
        navigate(`/${path}`);
      } else {
        setIsLoginModal(true);
      }
    },
    [authState]
  );

  return (
    <>
      <Wrapper>
        <img src={peachmong} alt="peachmong" />
        <div>Copyright © 2022 Peachmong All rights reserved.</div>
      </Wrapper>
      <WrapperMobile>
        <Item onClick={() => navigate('/')}>
          <Main width={23} height={23} fill={'#FF9A62'} />

          <div>둘러보기</div>
        </Item>
        <Item onClick={() => goPage('Wishlist')}>
          <Wishlist width={23} height={23} fill={'#FF9A62'} />
          <div>위시리스트</div>
        </Item>
        <Item onClick={() => goPage('Account')}>
          <Account width={23} height={23} fill={'#FF9A62'} />
          <div>프로필</div>
        </Item>
      </WrapperMobile>
    </>
  );
};

export default Footer;
