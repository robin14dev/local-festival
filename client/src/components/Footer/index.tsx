import React, { useContext, useState, useEffect } from "react";
import peachmong from "../../assets/peachmong.png";
import { useNavigate } from "react-router-dom";
import { ReactComponent as Account } from "../../assets/profile.svg";
import { ReactComponent as Wishlist } from "../../assets/heart-empty.svg";
import { ReactComponent as Main } from "../../assets/search.svg";
import { UserContext } from "../../contexts/userContext";
import { LoginModalDispatchContext } from "../../contexts/LoginModalContext";
import { Container, MobileContainer, Item } from "./styled";

const Footer = () => {
  const [isMobile, setIsMobile] = useState(window.innerWidth <= 480);
  let navigate = useNavigate();
  const { authState } = useContext(UserContext);
  const setIsLoginModal = useContext(LoginModalDispatchContext);
  const goPage = (path: string) => {
    if (authState.loginStatus) {
      navigate(`/${path}`);
    } else {
      setIsLoginModal(true);
    }
  };

  useEffect(() => {
    const handleResize = () => {
      setIsMobile(window.innerWidth <= 480);
    };

    window.addEventListener("resize", handleResize);

    return () => {
      window.removeEventListener("resize", handleResize);
    };
  }, []);
  return (
    <>
      {isMobile ? (
        <MobileContainer data-testid="Footer-mobile">
          <Item onClick={() => navigate("/")}>
            <Main width={23} height={23} fill={"#FF9A62"} />

            <div>둘러보기</div>
          </Item>
          <Item onClick={() => goPage("Wishlist")}>
            <Wishlist width={23} height={23} fill={"#FF9A62"} />
            <div>위시리스트</div>
          </Item>
          <Item onClick={() => goPage("Account")}>
            <Account width={23} height={23} fill={"#FF9A62"} />
            <div>프로필</div>
          </Item>
        </MobileContainer>
      ) : (
        <Container data-testid="Footer">
          <img src={peachmong} alt="peachmong" />
          <div>Copyright © 2022 Peachmong All rights reserved.</div>
        </Container>
      )}
    </>
  );
};

export default Footer;
