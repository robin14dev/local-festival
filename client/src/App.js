import React, { useEffect, useState } from "react";
import Header from "./components/Header";
import Mainpage from "./pages/Mainpage";
import MyPick from "../src/pages/MyPick";
import Footer from "./components/Footer";
import { Routes, Route } from "react-router-dom";
import "./App.css";
import Detailviewpage from "./pages/Detailviewpage";
import styled from "styled-components";
import axios from "axios";
import Signup from "./components/Signup";
import AccountSetting from "./pages/AccountSetting";
const Wrapper = styled.div`
  width: 100%; //1425px 스크롤바 생김
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  margin: 0 ;
  padding: 0;
`;

function App() {
  const [authState, setAuthState] = useState({
    userId: "",
    account:"",
    nickname: "",
    loginStatus: false,
  });
  const [festivalData, setFestivalData] = useState(null);
  const [pickItems, setPickItems] = useState([]);
  const [filteredData, setFilteredData] = useState(festivalData);

  // const infiniteScroll = (addData) => {
  //   //console.log('addData', addData);
  //   // console.log('here');
  //   console.log(festivalData, addData);
  //   let addToData = festivalData.concat(addData)
  //   console.log('new state',addToData);
  //   //alert('here')
  //   setFestivalData(addToData)
  //   setFilteredData(addToData)
  // }
 
  const loginHandler = ( userId,account,nickname, loginStatus) => {
    // isLogin ? setIsLogin(false) : setIsLogin(true);
    // console.log(nickname, userId, loginStatus);
    //* 로그인한 후의 유저정보 상태변경입니다.
    const nextState = {
      userId: userId,
      account: account,
      nickname: nickname,
      loginStatus: loginStatus,
    };
    setAuthState(nextState);
    //console.log("나 실행???");

    //# 유저별 찜한 축제 가져오기
    axios.get(`${process.env.REACT_APP_SERVER_ADDRESS_LOCAL}/pick`
    ,{headers: {
      accesstoken: sessionStorage.getItem("accesstoken"),
    }}
    ).then((response) => { 
      const pickedFestivalId = response.data;
      //console.log(pickedFestivalId);
      // const festivalIdArr = pickedFestivalId.map(ele => ele.local_id)
      // const pickedFestivalByUser = festivalData.filter(ele => festivalIdArr.indexOf(ele.id) > -1)
      //{festivalId: 4} 
      setPickItems(pickedFestivalId);
    });
  };
  const onSearch = (searchText) => {

    //에러 : 경상도라고 치면 안나옴 => 경상남도, 경상북도 다 나오게
    // 6월 
    // 서울시,인천시,부산시,광주시,대전시,울산시,대구시 => '시' 빼기
    // 경상도, 전라도, 충청도 => '도' 빼기
    // * 추후 수정 : 무한스크롤 구현 이후, 렌더링 되기전의 축제들을 검색했을때도 나와야 하므로 이때는 db에서 조회해야됨.
    // * 스크롤 하면서 넘어온 정보들 안에 원하는 검색결과가 있으면 클라이언트에서 보내주고, 없으면 db에 요청

    console.log(searchText);
    // setCondition(searchText);
    const filteredFestival = festivalData.filter(
      (festival) =>
        festival.location.includes(searchText) ||
        festival.title.includes(searchText) ||
        festival.location.includes(searchText)
    );

    setFilteredData(filteredFestival);
    //alert(`${filteredFestival.length}개의 축제가 진행중입니다.`)
    // }
  };

  const resetCondition = () => {
    setFilteredData(festivalData);
  };

  const togglePick = (newPick) => {
    //#1. 픽했는지 아닌지 부터 확인
    console.log('festival', newPick);
    const found = pickItems.filter((el) => el.festivalId === newPick.festivalId);
    console.log('wht is found', found);
    if (found.length !== 0) { // 이미 찜목록에 있으면 해제를 시켜줘야됨
      console.log("found");

    //#2-1. 픽 해제해서 서버에 픽 해제한 정보 보내주기
      console.log("removeId what!!!", newPick);

      //*서버에 삭제요청 보내기
      axios.delete(`${process.env.REACT_APP_SERVER_ADDRESS_LOCAL}/pick`,  {data : {festivalId: newPick.festivalId}, headers: {
        accesstoken: sessionStorage.getItem("accesstoken"),
      }})
      .then(response => {
        console.log(response.data.message);
      })
      .catch(err => {
        console.log(err);
      })

        setPickItems(pickItems.filter((el) => el.festivalId !== newPick.festivalId));
    } else {
      console.log("add new");
     //#2-2. 픽해서 서버에 픽한 정보 보내주기
      axios
        .post(`${process.env.REACT_APP_SERVER_ADDRESS_LOCAL}/pick`, {
          festivalId: newPick.festivalId,
        }, {headers: {
          accesstoken: sessionStorage.getItem("accesstoken"),
        }})
        .then((response) => {
          console.log(response.data.message);
        })
        .catch((err) => {
          console.log(err);
        });

      setPickItems([
        ...pickItems,
        newPick
      ]);
    }
  };
 
  const handleAuthState = (nickname)=>{
    const nextAuthState = authState
    nextAuthState.nickname = nickname
    setAuthState(nextAuthState)
   
  }

  
 

  useEffect(() => {
   
    const fetchData = async () => {
      try {
        const response = await axios.get(
          `${process.env.REACT_APP_SERVER_ADDRESS_LOCAL}/festivals`,
          {params: {limit: 10}}, 
         
        );
        if (response) {
          setFestivalData(response.data);
          setFilteredData(response.data);
        } else {
          console.log("no fetch data & use dummyData");
          setFestivalData([]);
        }
        // setFestivalData(dummyData);
      } catch (error) {
        console.log(error);
        console.log(
          "error났을때 client에 저장해놓은 dummydata(20개짜리)로 렌더링합니다"
        );
        setFestivalData([]);
      }
    };
    fetchData();
    
    if(sessionStorage.getItem("accesstoken")){
      axios.get(`${process.env.REACT_APP_SERVER_ADDRESS_LOCAL}/users`, {
        headers: {
          accesstoken: sessionStorage.getItem("accesstoken"),
        },
      })
      .then(response => {
        //console.log(response.data);
        const {userId,account, nickname} = response.data.data
       
        setAuthState({
          userId :userId,
          account:account,
          nickname : nickname,
          loginStatus : true
  
        })
  
        //* 새로고침시 유저가 픽한 상태도 유지되야 하므로
        axios.get(`${process.env.REACT_APP_SERVER_ADDRESS_LOCAL}/pick`, {headers: {
        accesstoken: sessionStorage.getItem("accesstoken"),
      }}
      ).then((response) => {
        //console.log(response.data.data);
        const pickedFestival = response.data;
        //console.log(pickedFestivalId);
        // const festivalIdArr = pickedFestivalId.map(ele => ele.local_id)
        // const pickedFestivalByUser = festivalData.filter(ele => festivalIdArr.indexOf(ele.id) > -1)
        //{festivalId: 4}
        setPickItems(pickedFestival);
      });
    
        
      })
      .catch(err => {
        console.log(err);
        console.log('에러 발생');
      })
    }


    
  }, []);

  
  return (
    <Wrapper>
      <Header loginHandler={loginHandler} authState={authState} />
      <Routes>
        <Route
          exact
          path="/"
          element={
            <Mainpage
              authState={authState}
              togglePick={togglePick}
              onSearch={onSearch}
              filteredData={filteredData}
              pickItems={pickItems}
              resetCondition={resetCondition}
              // infiniteScroll={infiniteScroll}
            />
          }
        ></Route>
        <Route
          exact
          path="/MyPick"
          element={
            <MyPick
              authState={authState}
              handleAuthState={handleAuthState}
              festivalData={festivalData}
              pickItems={pickItems}
              togglePick={togglePick}
            />
          }
        ></Route>
        <Route
          exact
          path="/Detailviewpage/festivalId/:id"
          element={<Detailviewpage  
            pickItems={pickItems} 
            togglePick={togglePick} 
            authState={authState}
            />}
        ></Route>
        <Route
        exact
        path="/AccountSetting"
        element = {
          <AccountSetting handleAuthState={handleAuthState} authState={authState}/>
        }
        >
          
        </Route>
        <Route exact path="/Signup" element={<Signup />}></Route>
      </Routes>
      <Footer />
    </Wrapper>
  );
}

export default App;
