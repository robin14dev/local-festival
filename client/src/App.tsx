import "./App.css";
import "../src/styles/common.scss";
import theme from "./styles/theme";
import React, { useState, useRef, useCallback } from "react";
import { Routes, Route } from "react-router-dom";
import { Helmet } from "react-helmet";
import styled, { ThemeProvider } from "styled-components";
import axios from "axios";
import { UserProvider } from "./contexts/userContext";
import { ModalContext } from "./contexts/modalContext";
import Wishlist from "./pages/Wishlist";
import DetailView from "./pages/DetailView";
import Main from "./pages/Main";
import Account from "./pages/Account";
import Header from "./components/Header";
import Footer from "./components/Footer";
import LoginModal from "./components/account/LoginModal";

const Wrapper = styled.div`
  width: 100%;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  margin: 0;
  padding: 0;
`;

const getUserPicksFromStorage: () => FestivalItem[] = () => {
  if (sessionStorage.getItem("picks")) {
    const picks = sessionStorage.getItem("picks");
    if (picks) {
      return JSON.parse(picks);
    }
  } else {
    return [];
  }
};

function App() {
  const [festivalData, setFestivalData] = useState<FestivalItem[]>([]);
  const [pickItems, setPickItems] = useState<FestivalItem[]>(
    getUserPicksFromStorage()
  );
  const [filteredData, setFilteredData] =
    useState<FestivalItem[]>(festivalData);
  const [isLoginModal, setIsLoginModal] = useState(false);
  const offset = useRef(0);

  const togglePick: togglePick = useCallback(
    async (newPick) => {
      //#1. 픽했는지 아닌지 부터 확인
      const found = pickItems.filter(
        (el) => el.festivalId === newPick.festivalId
      );
      if (found.length !== 0) {
        // 이미 찜목록에 있으면 해제를 시켜줘야됨
        //픽 해제해서 서버에 픽 해제한 정보 보내주기
        try {
          await axios.delete(`${process.env.REACT_APP_SERVER_URL}/pick`, {
            data: { festivalId: newPick.festivalId },
            headers: {
              accesstoken: sessionStorage.getItem("accesstoken") ?? "",
            },
          });
          const nextPicks = pickItems.filter(
            (item) => item.festivalId !== newPick.festivalId
          );
          setPickItems(nextPicks);
          sessionStorage.setItem("picks", JSON.stringify(nextPicks));
        } catch (error) {
          console.log(error);
        }
      } else {
        //픽해서 서버에 픽한 정보 보내주기
        try {
          await axios.post(
            `${process.env.REACT_APP_SERVER_URL}/pick`,
            {
              festivalId: newPick.festivalId,
            },
            {
              headers: {
                accesstoken: sessionStorage.getItem("accesstoken") ?? "",
              },
            }
          );
          const nextPicks = [newPick, ...pickItems];

          setPickItems(nextPicks);
          sessionStorage.setItem("picks", JSON.stringify(nextPicks));
        } catch (error) {
          console.log(error);
        }
      }
    },
    [pickItems]
  );
  const getPickedItems = async () => {
    try {
      const result = await axios.get(
        `${process.env.REACT_APP_SERVER_URL}/pick`,
        {
          headers: {
            accesstoken: sessionStorage.getItem("accesstoken") ?? "",
          },
        }
      );

      setPickItems(result.data);
      sessionStorage.setItem("picks", JSON.stringify(result.data));
    } catch (error) {
      console.log(error);
    }
  };
  return (
    <ThemeProvider theme={theme}>
      <ModalContext.Provider value={{ isLoginModal, setIsLoginModal }}>
        <UserProvider>
          <Wrapper>
            <Helmet>
              <title>이번주엔 어디로 가볼까? - LOCO</title>
            </Helmet>
            {isLoginModal && <LoginModal setIsLoginModal={setIsLoginModal} />}

            <Header />
            <Routes>
              <Route
                path="/"
                element={
                  <Main
                    togglePick={togglePick}
                    filteredData={filteredData}
                    pickItems={pickItems}
                    setFestivalData={setFestivalData}
                    setFilteredData={setFilteredData}
                    offset={offset}
                  />
                }
              ></Route>
              <Route
                path=":search"
                element={
                  <Main
                    togglePick={togglePick}
                    filteredData={filteredData}
                    pickItems={pickItems}
                    setFestivalData={setFestivalData}
                    setFilteredData={setFilteredData}
                    offset={offset}
                  />
                }
              ></Route>

              <Route
                path="/Wishlist"
                element={
                  <Wishlist pickItems={pickItems} togglePick={togglePick} />
                }
              ></Route>
              <Route
                path="/Detail/:festivalId/*"
                element={<DetailView togglePick={togglePick} />}
              ></Route>
              <Route path="/Account" element={<Account />}></Route>
            </Routes>
            <Footer setIsLoginModal={setIsLoginModal} />
          </Wrapper>
        </UserProvider>
      </ModalContext.Provider>
    </ThemeProvider>
  );
}

export default App;
