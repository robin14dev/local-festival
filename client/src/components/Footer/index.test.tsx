import React from "react";
import { MemoryRouter } from "react-router-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import App from "../../App";
import Footer from ".";
import { UserContext } from "../../contexts/userContext";
import { LoginModalContextProvider } from "../../contexts/LoginModalContext";
import {
  URLTestComponent,
  intersectionObserverMock,
  mockAuthState,
  mockSetAuthState,
} from "../../test/mock";

beforeEach(() => {
  window.IntersectionObserver = jest
    .fn()
    .mockImplementation(intersectionObserverMock);
});

describe("<Footer/> 테스트", () => {
  it("모바일 너비에 따라 컴포넌트가 성공적으로 렌더링 됩니다", () => {
    // const mediaQueryList = window.matchMedia("(min-width: 476px)");

    global.innerWidth = 480;
    const { container } = render(
      <MemoryRouter initialEntries={["/"]}>
        <Footer />
      </MemoryRouter>
    );
    expect(screen.getByTestId("Footer-mobile")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("데스크탑 너비에 따라 컴포넌트가 성공적으로 렌더링 됩니다", () => {
    global.innerWidth = 481;
    const { container } = render(
      <MemoryRouter initialEntries={["/"]}>
        <Footer />
      </MemoryRouter>
    );

    expect(screen.getByTestId("Footer")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("모바일 UI일 경우, [둘러보기]를 클릭하면 메인 페이지로 이동합니다", () => {
    global.innerWidth = 480;

    const { container } = render(
      <MemoryRouter initialEntries={["/Wishlist"]}>
        <UserContext.Provider
          value={{
            authState: mockAuthState.login,
            setAuthState: mockSetAuthState,
          }}
        >
          <URLTestComponent />
          <App />
        </UserContext.Provider>
      </MemoryRouter>
    );
    expect(screen.getByTestId("WishlistPage")).toBeInTheDocument();
    const gotoMainpageBtn = screen.getByText("둘러보기");
    const pathName = screen.getByText("pathname is /Wishlist");
    expect(gotoMainpageBtn).toBeInTheDocument();
    fireEvent.click(gotoMainpageBtn);
    expect(screen.getByTestId("MainPage")).toBeInTheDocument();
    expect(pathName.textContent).toBe("pathname is /");
  });
  it("모바일 UI일 경우, 로그인 상태일 때, [위시리스트]를 클릭하면 위시리스트 페이지로 이동합니다", () => {
    global.innerWidth = 480;

    const { container } = render(
      <MemoryRouter initialEntries={["/"]}>
        <UserContext.Provider
          value={{
            authState: mockAuthState.login,
            setAuthState: mockSetAuthState,
          }}
        >
          <URLTestComponent />
          <App />
        </UserContext.Provider>
      </MemoryRouter>
    );

    expect(screen.getByTestId("MainPage")).toBeInTheDocument();
    const pathName = screen.getByText("pathname is /");
    const button = screen.getByText("위시리스트");
    expect(button).toBeInTheDocument();

    fireEvent.click(button);
    expect(screen.getByTestId("WishlistPage")).toBeInTheDocument();
    expect(pathName.textContent).toBe("pathname is /Wishlist");
  });
  it("모바일 UI일 경우, 로그인 상태일 때, [프로필]을 클릭하면 계정 페이지로 이동합니다", () => {
    global.innerWidth = 480;

    const { container } = render(
      <MemoryRouter initialEntries={["/"]}>
        <UserContext.Provider
          value={{
            authState: mockAuthState.login,
            setAuthState: mockSetAuthState,
          }}
        >
          <URLTestComponent />
          <App />
        </UserContext.Provider>
      </MemoryRouter>
    );

    expect(screen.getByTestId("MainPage")).toBeInTheDocument();
    const pathName = screen.getByText("pathname is /");
    const button = screen.getByText("프로필");
    expect(button).toBeInTheDocument();

    fireEvent.click(button);
    expect(screen.getByTestId("AccountPage")).toBeInTheDocument();
    expect(pathName.textContent).toBe("pathname is /Account");
  });
  it("모바일 UI일 경우, 로그아웃 상태일 때, [위시리스트] 또는 [프로필]을  클릭하면 로그인 모달이 렌더링 됩니다", () => {
    global.innerWidth = 480;

    const { container } = render(
      <MemoryRouter initialEntries={["/"]}>
        <LoginModalContextProvider>
          <UserContext.Provider
            value={{
              authState: mockAuthState.logout,
              setAuthState: mockSetAuthState,
            }}
          >
            <URLTestComponent />
            <App />
          </UserContext.Provider>
        </LoginModalContextProvider>
      </MemoryRouter>
    );

    expect(screen.getByTestId("MainPage")).toBeInTheDocument();
    const pathName = screen.getByText("pathname is /");
    const button = screen.getByText("프로필");
    expect(button).toBeInTheDocument();

    fireEvent.click(button);
    expect(screen.getByTestId("LoginModal")).toBeInTheDocument();
    expect(pathName.textContent).toBe("pathname is /");
  });
});
