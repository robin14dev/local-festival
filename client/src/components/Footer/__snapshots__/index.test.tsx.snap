// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> 테스트 데스크탑 너비에 따라 컴포넌트가 성공적으로 렌더링 됩니다 1`] = `
<div>
  <footer
    class="sc-dsHJmm dhTnQP"
    data-testid="Footer"
  >
    <img
      alt="peachmong"
      src="peachmong.png"
    />
    <div>
      Copyright © 2022 Peachmong All rights reserved.
    </div>
  </footer>
</div>
`;

exports[`<Footer/> 테스트 모바일 너비에 따라 컴포넌트가 성공적으로 렌더링 됩니다 1`] = `
<div>
  <footer
    class="sc-cZFQFd hkLmQD"
    data-testid="Footer-mobile"
  >
    <div
      class="sc-jTjUTQ VPFIK"
    >
      <svg
        fill="#FF9A62"
        height="23"
        width="23"
      >
        search.svg
      </svg>
      <div>
        둘러보기
      </div>
    </div>
    <div
      class="sc-jTjUTQ VPFIK"
    >
      <svg
        fill="#FF9A62"
        height="23"
        width="23"
      >
        heart-empty.svg
      </svg>
      <div>
        위시리스트
      </div>
    </div>
    <div
      class="sc-jTjUTQ VPFIK"
    >
      <svg
        fill="#FF9A62"
        height="23"
        width="23"
      >
        profile.svg
      </svg>
      <div>
        프로필
      </div>
    </div>
  </footer>
</div>
`;
